### PR TITLE
add ISSUE_TEMPLATE for github and xr.show_versions()

### PR DESCRIPTION
**Pull request by jhamman**
- [x] xref #986
 - [x] Tests added / passed
 - [x] Passes ``git diff upstream/master | flake8 --diff``
 - [x] Fully documented, including `whats-new.rst` for all changes and `api.rst` for new API

This PR adds a new module level function `xr.show_versions()` and a new Github Issue Template to help with debugging user issues. Hopefully, we stop having to ask "what version of dask are you using?" Functionality is copied from [pandas](https://github.com/pandas-dev/pandas/blob/e99f56e43ab65710a34440067efe372d5f74280b/pandas/util/_print_versions.py).

### CODE REVIEW

**Comment by fmaussion** [comment]
This is a good idea! 

But it is going to be a mess until a new version of xarray is released, since users won't have access to the utility function...

**Comment by jhamman** [comment]
> But it is going to be a mess until a new version of xarray is released, since users won't have access to the utility function...

Good point. I guess we could hold off on merging for a bit then.

File: xarray/util/print_versions.py [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+
+def show_versions(as_json=False):
+    sys_info = get_sys_info()
+
+    deps = [
+        # (MODULE_NAME, f(mod) -> mod version)
+        ("xarray", lambda mod: mod.__version__),
+        ("pandas", lambda mod: mod.__version__),
+        ("numpy", lambda mod: mod.version.version),
+        ("scipy", lambda mod: mod.version.version),
```
Comment (shoyer) [inline xarray/util/print_versions.py]: Can we just use `mod.__version__` for numpy and scipy? That's the more standard way to access this information.

Comment (jhamman) [inline xarray/util/print_versions.py]: yes, that works. Done.

**Comment by jhamman** [comment]
Let's just wait until we have our next release.

**Pull request merged**